File: sort/insertionSort.js
function insertionSort(A) {
  for (j = 1; j < A.length; j++) {
    let value = A[j];
    let i = j - 1;
    while (i >= 0 && A[i] > value) {
      A[i + 1] = A[i];
      i--;
    }
    A[i + 1] = value;
  }
  return A;
}

console.log(insertionSort([5, 1, 4, 7, 9, 3]));

// O(n^2)

File: algorithms/sort/insertionSort.py
def insertionSort(A):
    for i in range(1, len(A)):
        value = A[i]
        j = i-1
        while j >= 0 and A[j] > value:
            A[j+1] = A[j]
            j -= 1
        A[j+1] = value
    return A


print(insertionSort([5, 6, 2, 4, 1]))

# O(n^2)
